server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -746,16 +774,29 @@ Batch nextBatch(SortedKeyValueIterator<Key,Value> iter, Range range, int num, Se
       iter.next();
     }
 
-    if (iter.hasTop() == false) {
-      endOfTabletReached = true;
-    }
+    if (yield.hasYielded()) {
+      continueKey = new Key(yield.getPositionAndReset());
+      skipContinueKey = true;
+      if (!range.contains(continueKey)) {
+        throw new IOException("Underlying iterator yielded to a position outside of its range: " + continueKey + " not in " + range);
+      }
 
-    if (endOfTabletReached) {
-      continueKey = null;
-    }
+      log.debug("Scan yield detected at position " + continueKey);
```

Comment:
>  If you want to know how often yielding is happening, you should look at the metrics you implemented and not place extra cruft in the log to just be removed in every other situation.

Do you happen to know if the Hadoop Metrics2 framework picks up configuration changes at runtime like the logging subsystem?